syntax = "proto3";

service Lamp {
    rpc change_state (ChangeStateRequest) returns (Response) {}
    rpc get_state (GetStateRequest) returns (Response) {}
    rpc change_color (ColorRequest) returns (Response) {}
}

message GetStateRequest {

}

message ColorRequest {
    optional string hex_color = 1;
}

message ChangeStateRequest {
    optional bool state = 1;
}

message Response {
    optional bool status = 1;
    optional string message = 2;
}